static/src/javascripts/projects/common/modules/commercial/create-ad-slot.js

```diff
@@ -1,131 +1,153 @@
+                ),
+                tablet: compile(
+                    adSizes.outOfPage,
+                    adSizes.mpu,
+                    config.switches.fluidAdverts ? adSizes.fluid : null
+                )
             }
         },
         inline: {
             sizeMappings: {
-                mobile:             '1,1|300,250',
-                'mobile-landscape': '1,1|300,250',
-                tablet:             '1,1|300,250'
+                mobile: compile(adSizes.outOfPage, adSizes.mpu)
             }
         },
         mostpop: {
             sizeMappings: {
-                mobile:             '1,1|300,250',
-                'mobile-landscape': '1,1|300,250',
-                tablet:             '1,1|300,250'
+                mobile: compile(adSizes.outOfPage, adSizes.mpu)
             }
         },
         'merchandising-high': {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|88,87'
+                mobile: compile(adSizes.outOfPage, adSizes.merchandising)
             }
         },
         spbadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         adbadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         fobadge: {
             label: false,
             refresh: false,
             sizeMappings: {
-                mobile: '1,1|140,90'
+                mobile: compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         comments: {
             sizeMappings: {
-                mobile:  '1,1|300,250'
+                mobile:  compile(adSizes.outOfPage, adSizes.badge)
             }
         },
         'top-above-nav': {
             sizeMappings: {
-                desktop: '1,1|88,70|728,90|940,230|900,250|970,250|' + fabricTopSlot
+                desktop: compile(
+                    adSizes.outOfPage,
+                    adSizes.fluid250,
+                    adSizes.leaderboard,
+                    adSizes.cascase,
+                    adSizes.superHeader,
+                    adSizes.billboard,
+                    adSizes.fabric
+                )
             }
         }
     };
 
-    function createAdSlotElement(name, attrs, classes) {
-        return $(document.createElement('div'))
-            .attr({
-                'id': 'dfp-ad--' + name,
-                'data-link-name': 'ad slot ' + name,
-                'data-test-id': 'ad-slot-' + name,
-                'data-name': name
-            })
-            .attr(attrs)
-            .addClass('js-ad-slot ad-slot ad-slot--dfp ' + classes.join(' '));
+    function compile(size1) {
+        var result = size1;
```

Comment:
Or, you could just use lodash's `toArray` helper and join the result:

```
function compile() {
    return toArray(arguments).join('|');
}
```

Depends how you feel about adding lodash dependencies, I suppose.